framework/src/media/MediaRecorderImpl.cpp

```diff
@@ -387,19 +395,14 @@ recorder_result_t MediaRecorderImpl::setVolume(int vol)
 	return ret;
 }
 
-void MediaRecorderImpl::setRecorderVolume(int vol, recorder_result_t& ret)
+void MediaRecorderImpl::setRecorderVolume(uint8_t vol, recorder_result_t& ret)
 {
 	medvdbg("setRecorderVolume\n");
 
-	int maxVolume = get_max_audio_volume();
-	if (vol < 0 || vol > maxVolume) {
-		meddbg("Volume is out of range vol : %d (recorder ranage 0 ~ %d)\n", vol, maxVolume);
-		return notifySync();
-	}
-
-	audio_manager_result_t result = set_input_audio_volume((uint8_t)vol);
+	audio_manager_result_t result = set_input_audio_gain(vol);
 	if (result != AUDIO_MANAGER_SUCCESS) {
 		meddbg("set_input_audio_volume failed vol : %d ret : %d\n", vol, result);
+		ret = RECORDER_ERROR;
```

Comment:
Based on policy what you decided, below is correct I think
```
if (result != AUDIO_MANAGER_SUCCESS) {
    if (result == AUDIO_MANAGER_DEVICE_NOT_SUPPORT) {
        ret = RECORDER_ERROR_INVALID_OPERATION;
    } else {
        ret = RECORDER_ERROR;
    }
}
```
But  you think this error can be expanded, switch-case is more good option

Reply:
I'll expand later. I think your suggestion is good enough for now.